File: android/src/main/kotlin/co/quis/flutter_contacts/crud/models/properties/PropertyMetadata.kt
package co.quis.flutter_contacts.crud.models.properties

data class PropertyMetadata(
    val dataId: String? = null,
    val rawContactId: String? = null,
) {
    companion object {
        fun fromJson(json: Map<String, Any?>): PropertyMetadata =
            PropertyMetadata(
                dataId = json["dataId"] as? String,
                rawContactId = json["rawContactId"] as? String,
            )
    }

    fun toJson(): Map<String, Any?> {
        val result = mutableMapOf<String, Any?>()
        dataId?.let { result["dataId"] = it }
        rawContactId?.let { result["rawContactId"] = it }
        return result
    }
}

File: ios/flutter_contacts/Sources/flutter_contacts/crud/models/properties/PropertyMetadata.swift
import Foundation

struct PropertyMetadata: Equatable, Hashable {
    let dataId: String?

    func toJson() -> Json {
        var json: Json = [:]
        json.set("dataId", dataId)
        return json
    }

    static func fromJson(_ json: Json) -> PropertyMetadata {
        PropertyMetadata(dataId: json["dataId"] as? String)
    }
}
